airflow/timetables/base.py

```diff
@@ -281,8 +282,17 @@ def generate_run_id(
         self,
         *,
         run_type: DagRunType,
-        logical_date: DateTime,
+        logical_date: DateTime | None,
         data_interval: DataInterval | None,
+        run_after: DateTime = timezone.coerce_datetime(timezone.utcnow()),
```

Comment:
Let’s not change this signature now. For cases with None logical_date, let’s just do

```python
logical_date=logical_date or run_after
```

to keep things work.